### PR TITLE
Jakarta EE 10 - Platform/API/RI support

### PR DESCRIPTION
**Pull request by matthiasblaesing**
This PR was created on the basis of https://github.com/apache/netbeans/pull/4692

The changes from that PR were rebased onto master and then squashed as far as it seemed reasonable. To check the valid rebase without changing the code, this was done:

```bash
git checkout -b pr-4692-ref master
git merge github/pr/4692
git checkout -b pr-4692 github/pr/4692
git rebase master
# + multiple squash rounds
git diff pr-4692-ref..pr-4692
```

results in

```diff
diff --git a/enterprise/websvc.restapi/nbproject/org-netbeans-modules-websvc-restapi.sig b/enterprise/websvc.restapi/nbproject/org-netbeans-modules-websvc-restapi.sig
index 9f7f657d6276..60e477dcc2d0 100644
--- a/enterprise/websvc.restapi/nbproject/org-netbeans-modules-websvc-restapi.sig
+++ b/enterprise/websvc.restapi/nbproject/org-netbeans-modules-websvc-restapi.sig
@@ -1,5 +1,5 @@
 #Signature file v4.1
-#Version 1.54
+#Version 1.55
 
 CLSS public abstract interface java.io.Serializable
 
diff --git a/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps b/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps
index 1f39e97a714e..5673dc068035 100644
--- a/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps
+++ b/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps
@@ -123,4 +123,4 @@ platform/o.apache.commons.lang3/external/commons-lang3-3.8.1.jar java/maven.embe
 ide/db.metadata.model/external/mysql-connector-j-8.0.31.jar ide/db.mysql/external/mysql-connector-j-8.0.31.jar
 
 # Jakarta EE API doc is repackaged to align with old NetBeans format, embedded indexes are thus identically and can be ignored.
-enterprise/jakartaee10.platform/external/generated-jakarta.jakartaee-api-10.0.0-javadoc.jar enterprise/jakartaee10.platform/external/jakarta.jakartaee-api-10.0.0-javadoc.jar
\ No newline at end of file
+enterprise/jakartaee10.platform/external/generated-jakarta.jakartaee-api-10.0.0-javadoc.jar enterprise/jakartaee10.platform/external/jakarta.jakartaee-api-10.0.0-javadoc.jar
```

### CODE REVIEW

**Review by mbien — APPROVED** [review]

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV7_0_0.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    <library id="Jersey 3">
+        <classpath>
+            <fileset dir="modules">
+                <include name="jackson.+\.jar"/>
+                <include name="jersey.+\.jar"/>
+                <include name="jettison.*\.jar"/>
+            </fileset>
+        </classpath>
+        <javadocs>
+            <link url="https://repo1.maven.org/maven2/org/glassfish/jersey/jersey-documentation/3.0.8/jersey-documentation-3.0.8-docbook.zip"/>
```
Comment (pepness) [inline enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV7_0_0.xml:77]: Jersey 3.0.x is for Jakarta EE 9, the url for Jakarta EE 10 is: 
https://repo1.maven.org/maven2/org/glassfish/jersey/jersey-documentation/3.1.0/jersey-documentation-3.1.0-docbook.zip

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV7_0_0.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<server>
+    <tools lib="lib">
+        <asadmin jar="client/appserver-cli.jar"/>
+    </tools>
+    <java version="18">
```
Comment (pepness) [inline enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV7_0_0.xml:26]: It supports up to version 19
[Release 7](https://github.com/eclipse-ee4j/glassfish/releases/tag/7.0.0)

**Comment by pepness** [comment]
Hi @matthiasblaesing, I made a [branch](https://github.com/pepness/incubator-netbeans/tree/pr-4692-ref) with some fixes and improvements like adding support for javadoc jars that only have the element-list file instead of the package-list file like is the case with the jakartaee-10 javadoc jar, without this the javadoc is not accessible within the IDE. I create a PR  to your branch but it has some newer PR that are already in master because I rebase it  with my master, sorry for the mess.

**Comment by matthiasblaesing** [comment]
Sorry for the confusion - @pepness I'm just building with your changes and push here to get testing and will review in parallel.

**Comment by matthiasblaesing** [comment]
Closing - force push to #4692 worked.

**Pull request closed without merging**